tlatools/org.lamport.tlatools/customBuild.xml

```diff
@@ -159,7 +150,9 @@
 			target="1.8"
 			encoding="UTF-8"
 		>
-			<!-- compilerarg value="-Xlint:deprecation"/-->
+			<!-- Suppress sun.misc.Unsafe warnings because Java can't allocate an
+					 array with more than 2 billion elements. -->
+			<compilerarg line="-XDignore.symbol.file=src/tlc2/tool/fp/LongArray.java" />
```

Comment:
Part of me thinks that the warnings about `Unsafe` are _good_, actually, and should be retained. But I think that adding `-Werror` is a noble goal, and what you've done here is probably the simplest solution in pursuit of that goal.

Reply:
The comment about 2b elements is only part of the story.  That could also be worked around by using an array of arrays like in MSBFPSet.  Using Unsafe also has the advantage that it moves the array out of the "control" of the garbage collector that won't be able to ever reclaim any memory anyway. 

The reason why it might make sense to migrate to the official successor API in recent JVMs is also direct support to allocates *zeroed* memory with `calloc` (currently a bottleneck in our LongArray).  I once wrote a JVM patch that exposed calloc in Unsafe.  I went as far as proposing it on the valhalla mailinglist, but the timing for its inclusion wasn't right.

Reply:
I do like keeping the warnings around because they're good warnings, and even actionable if we can upgrade to the new API we talked about in the community chat last week, but they also make it difficult to find actual compiler errors in the text that is committed.

Reply:
Please amend the comment to mention that LongArray also doesn't incur GC cost.

Reply:
I just changed the comment there to reflect the silencing of the warnings itself; reasons for why the API is being used can go in the LongArray.java file.